pkg/storage/s3/getter.go

```diff
@@ -18,16 +20,13 @@ func (s *Storage) Info(ctx context.Context, module, version string) ([]byte, err
 	const op errors.Op = "s3.Info"
 	ctx, span := observ.StartSpan(ctx, op.String())
 	defer span.End()
-	exists, err := s.Exists(ctx, module, version)
```

Comment:
If we do not ensure these checks; will this not break in private networks where modules are already cached; but cannot fallback.

Reply:
If the modules are already cached, the `s.open` below will succeed; if not, this method will return the same `errors.KindNotFound` like before. From the caller of this method, there is no difference.